src/buildstream/_scheduler/job.py

```diff
@@ -322,111 +271,23 @@ async def _parent_child_completed(self, returncode):
         else:
             status = JobStatus.FAIL
 
-        self.parent_complete(status, self._result)
+        self._complete_cb(self, self._element, status, self._result)
```

Comment:
I'd suggest renaming the `_parent_child_completed()` method as well. Especially the 'parent' term no longer makes sense. Also the doc comments should be updated to refer to threads, not processes.

Reply:
I skipped this as after removing the element job, we're still using subclasses and overrides. Let's do the rename later